src/WebJobs.Script/Environment/EnvironmentExtensions.cs

```diff
@@ -254,9 +254,22 @@ public static bool IsWindowsAzureManagedHosting(this IEnvironment environment)
         /// </summary>
         /// <param name="environment">The environment to verify</param>
         /// <returns><see cref="true"/> if running in a Linux Consumption App Service app; otherwise, false.</returns>
-        public static bool IsLinuxConsumption(this IEnvironment environment)
+        public static bool IsAnyLinuxConsumption(this IEnvironment environment)
         {
-            return !environment.IsAppService() && !string.IsNullOrEmpty(environment.GetEnvironmentVariable(ContainerName));
+            return environment.IsLinuxConsumptionOnAtlas() || environment.IsLinuxConsumptionOnLegion();
+        }
+
+        public static bool IsLinuxConsumptionOnAtlas(this IEnvironment environment)
+        {
+            return !environment.IsAppService() &&
+                   !string.IsNullOrEmpty(environment.GetEnvironmentVariable(ContainerName)) &&
+                   string.IsNullOrEmpty(environment.GetEnvironmentVariable(LegionServiceHost));
+        }
+
+        public static bool IsLinuxConsumptionOnLegion(this IEnvironment environment)
+        {
+            return !string.IsNullOrEmpty(environment.GetEnvironmentVariable(WebsitePodName)) &&
```

Comment:
Should this also have !environment.IsAppService()?

Reply:
Also a case if ContainerName exists?

Reply:
I think LegionHost check should cover IsAppService check as well. basically they are mutually exclusive already

it is okay for a container to have both pod name and containername. the presence of legionhost appsetting is the deciding factor.